kernelci/cli/storage.py

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2023 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+
+"""Tool to interact with storage services supported by KernelCI"""
+
+import os
+
+import click
+
+import kernelci.config
+import kernelci.storage
+from . import Args, kci
+
+
+@kci.group(name='storage')
+def kci_storage():
+    """Interact with KernelCI storage services"""
+
+
+@kci_storage.command(secrets=True)
+@click.argument('filename', type=click.Path(exists=True))
+@click.argument('path', required=False)
+@Args.config
+@Args.storage
+def upload(filename, path, config, storage, secrets):
+    """Upload FILENAME to the designated storage service in PATH"""
+    configs = kernelci.config.load(config)
+    storage_config = configs['storage'][storage]
+    storage = kernelci.storage.get_storage(
+        storage_config, secrets.storage.credentials
+    )
+    url = storage.upload_single(
+        (filename, os.path.basename(filename)),
+        path or ''
+    )
+    click.echo(url)
```

Comment:
We could add a `--silent` or `--quiet` option... but what's the issue here?

Reply:
Yes but I would expect the output to be required and reused in that script, otherwise you can't include your just-uploaded artifact anywhere else.  Typically it would be stored in a variable and added to a node object.

Reply:
Yes this needs to output the URL, it's designed to be like this and it's a choice.  I know noise is not nice and Unix commands traditionally don't output anything when there's no error, but this is a conscious choice.  The previous implementation didn't output anything unless you passed `--verbose` and that cause lots of issues.